src/app/Recordings/Filters/DatetimeFilter.tsx

```diff
@@ -26,153 +27,239 @@ import {
   HelperTextItem,
   InputGroup,
   Popover,
-  PopoverPosition,
-  Stack,
-  StackItem,
   TextInput,
   ValidatedOptions,
   InputGroupItem,
+  InputGroupText,
+  CalendarProps,
 } from '@patternfly/react-core';
 import { OutlinedCalendarAltIcon, SearchIcon } from '@patternfly/react-icons';
-import { t } from 'i18next';
 import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { DATETIME_INPUT_MAXLENGTH } from './const';
 
-export interface DateTimeFilterProps {
-  onSubmit: (dateISO: string) => void;
-}
-
-const _emptyDatetimeInput: {
+interface _DatetimeState {
   text: string;
-  date: Date | undefined; // Ignore timezone
+  date?: Date; // Ignore timezone
   validation: ValidatedOptions;
-} = {
+}
+
+const _emptyDatetimeInput: _DatetimeState = {
   text: '',
   date: undefined,
   validation: ValidatedOptions.default,
 };
 
+export interface DateTimeRange {
+  from?: Date | string;
+  to?: Date | string;
```

Comment:
Can these be always `Date`, and wherever they are currently assigned `string`s be forced to do the conversion at that point?